### PR TITLE
Upgrade k8s dependencies to v1.26 release train

### PR DESCRIPTION
**Pull request by hjiawei**
## Description

This changeset upgrades k8s dependencies and utilities to the v1.26 release train. The test results are at [here](https://tigera.semaphoreci.com/workflows/e5770668-d606-41fd-be0b-5811f597484f?pipeline_id=f420d4e2-c220-448f-adf8-830342604b1b) which are similar to the current master status.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: apiserver/cmd/apiserver/server/options.go [hunk 1]
```diff
@@ -102,6 +102,9 @@ func (o *CalicoServerOptions) Config() (*apiserver.Config, error) {
 		}
 	}
 
+	if err := o.RecommendedOptions.Etcd.Complete(serverConfig.StorageObjectCountTracker, serverConfig.DrainedNotify(), serverConfig.AddPostStartHook); err != nil {
```
Comment (hjiawei) [inline apiserver/cmd/apiserver/server/options.go:105]: Complete must be called exactly once before using any of the Apply methods. This change is similar to the upstream kubernetes/apiserver [implementation](https://github.com/kubernetes/kubernetes/blob/release-1.26/staging/src/k8s.io/apiserver/pkg/server/options/recommended.go#L104).

File: apiserver/docker-image/Dockerfile.amd64 [hunk 2]
```diff
@@ -2,7 +2,7 @@ ARG UBI_IMAGE
 
 FROM ${UBI_IMAGE} as ubi
 
-RUN microdnf update
+RUN microdnf upgrade
```
Comment (hjiawei) [inline apiserver/docker-image/Dockerfile.amd64:5]: `update` is deprecated and an alias for `upgrade`. Use `upgrade` instead.

File: cni-plugin/pkg/install/install.go [hunk 3]
```diff
@@ -141,7 +140,7 @@ func Install() error {
 	c.ServiceAccountToken = make([]byte, 0)
 	var err error
 	if fileExists(serviceAccountTokenFile) {
-		log.Info("Running as a Kubernetes pod")
+		logrus.Info("Running as a Kubernetes pod")
```
Comment (hjiawei) [inline cni-plugin/pkg/install/install.go:143]: We use `logrus` in all our components as well as this file. `prometheus/common/log` is removed from the official package a while back.

File: felix/bpf/proxy/lb_src_range_test.go [hunk 4]
```diff
@@ -65,7 +65,7 @@ func testfn(makeIPs func(ips []string) proxy.K8sServicePortOption) {
 	}
 
 	state := proxy.DPSyncerState{
-		SvcMap: k8sp.ServiceMap{
+		SvcMap: k8sp.ServicePortMap{
```
Comment (hjiawei) [inline felix/bpf/proxy/lb_src_range_test.go:68]: `ServiceMap` is renamed to `ServicePortMap` in v0.26.

File: kube-controllers/pkg/controllers/node/controller.go [hunk 5]
```diff
@@ -125,7 +125,10 @@ func NewNodeController(ctx context.Context,
 	}
 
 	// Set the handlers on the informers.
-	nc.nodeInformer.AddEventHandler(nodeHandlers)
+	if _, err := nc.nodeInformer.AddEventHandler(nodeHandlers); err != nil {
```
Comment (hjiawei) [inline kube-controllers/pkg/controllers/node/controller.go:128]: Resolve static check errors (same for all other places).

**Review by mgleung — APPROVED** [review]
LGTM though I don't think that the failing node tests are solely flakes so I need to take a look at those.

**Review by mgleung — APPROVED** [review]

**Pull request merged**